Extend reward classifier for multiple camera views

michel-aractingi opened the pull request:
# What this does
(A) This PR adds the possibility to train the reward classifier with multiple camera images. The architecture includes:
1- one pretrained resnet encoder for all images (with frozen parameters)
2- each image is passed through the encoder to get a compressed representation.
3- The representations are concatenated and passed through an MLP that will be trained to predict the reward. 

(B) This PR also extends the `eval_on_robot.py` script to load the reward classifier and the label each timestep with the predicted reward in real-time. 

(C) in `lerobot/common/robot_devices/control_utils.py` a new function to reset the follower arm to the initial position it was in at the start of the control. This is important for collecting a dataset with rewards so we don't get redundant frames in the trajectory (related to manually reseting the robot) after the task has terminated.

# How to test
- Collect a dataset with your robot #528 .
- Train a reward classifier with `lerobot/scripts/train_hilserl_classifier.py`
- Test the reward classifier with `eval_on_robot.py`, example:
```
python lerobot/scripts/eval_on_robot.py \
    --robot-path lerobot/configs/robot/so100.yaml \
    --reward-classifier-pretrained-path outputs/classifier/checkpoints/best/pretrained_model \
    --reward-classifier-config-file lerobot/configs/policy/hilserl_classifier.yaml \
    --display-cameras 1
```

**NOTE** you can use a dataset I collected with so100 `aractingi/pick_place_lego_cube_1`

helper2424 reviewed (approved):
The part related to the reward classifier looks good, I have checked locally - it converges, the only one thing is failling test

`pytest tests/test_train_hilserl_classifier.py` I think that need to fix it before merging

michel-aractingi commented:
Thanks @helper2424 ! The test issue was due to the addition of multi-camera training. I updated the test files and added additional testing function for training with two camera sources.

helper2424 commented:
Looks great.
